app/api/dao/user.py

```diff
@@ -143,20 +145,22 @@ def get_user_by_username(username: str):
         return UserModel.find_by_username(username)
 
     @staticmethod
-    def list_users(user_id: int, search_query: str = "", is_verified = None):
+    def list_users(user_id: int, search_query: str = "", page: int = DEFAULT_PAGE, per_page: int = DEFAULT_USERS_PER_PAGE, is_verified = None):
         """ Retrieves a list of verified users with the specified ID.
         
         Arguments:
             user_id: The ID of the user to be listed.
             search_query: The search query for name of the users to be found.
             is_verified: Status of the user's verification; None when provided as an argument.
+            page: The page of users to be returned
+            per_page: The number of users to return per page
         
         Returns:
             A list of users matching conditions and the HTTP response code.
         
         """
 
-        users_list = UserModel.query.filter(UserModel.id != user_id).all()
+        users_list = UserModel.query.filter(UserModel.id != user_id).paginate(page, per_page, False).items
```

Comment:
Maybe we can think about this on another issue, but I worry about the filters you have in lines down this code (line 166 to 170). Because you get 10 items on line 163, then you filter, which can mean you return in the endpoint a number < `per_page` value. And it should not happen like that right? Not sure if this is to worry here.
Example for this: 20 users -> first 2 are not `is_verified` -> filter by `is_verified` -> return in the 1st page, 8 users (without `is_verified`) and then in the 2nd page you return the last 10.

Hope this makes sense 🤞

Reply:
@isabelcosta makes total sense. Do you have an idea on how to go about doing the `is_verified` and `search_query` in the database query line?

Reply:
@isabelcosta Update: I figured it out :) we now filter with `is_verified` and `search_query` during the DB query instead of waiting to filter until after the DB call. You can see the updates with these two commits: [one](https://github.com/anitab-org/mentorship-backend/pull/541/commits/b7727f5d7e6b83f863ebd77c62fc9f827e303cfb) and [two](https://github.com/anitab-org/mentorship-backend/pull/541/commits/4c298dc54e0c7c09a234f400ffda87cf75fb0ff9)